core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -127,6 +127,17 @@ case class GroupSummary(state: String,
                         members: List[MemberSummary])
 
 /**
+  * We cache offset commits along with their commit record offset. This enables us to ensure that the latest offset
+  * commit is always materialized when we have a mix of transactional and regular offset commits. Without preserving
+  * information of the commit record offset, compaction of the offsets topic it self may result in the wrong offset commit
+  * being materialized.
+  */
+case class CommitRecordMetadataAndOffset(commitRecordOffset: Long,
```

Comment:
One more comment to pile on: it might be a little safer to use `Option[Long]` to ensure we're doing the sentinel checking where we need to.